engines/router/missionctl/server/upi/server.go

```diff
@@ -40,9 +37,67 @@ func NewUPIServer(mc missionctl.MissionControlUPI) *Server {
 	}
 }
 
+// UniversalPredictionServiceServer is the internal server that communicates in bytes
+// It requires the codec to be passing []byte instead of a proto object. It is registered
+// the same way as the generated proto which has the following interface instead.
+// PredictValues(context.Context, *PredictValuesRequest) (*PredictValuesResponse, error)
+type UniversalPredictionServiceServer interface {
+	PredictValues(context.Context, []byte) ([]byte, error)
```

Comment:
This is the magic/hack as appears from the generated proto that its 
`PredictValues(ctx context.Context, req *upiv1.PredictValuesRequest) (*upiv1.PredictValuesResponse, error)`

But it is in fact this signature internally instead.
`PredictValues(context.Context, []byte) ([]byte, error)`